src/rai_extensions/rai_perception/rai_perception/components/topic_utils.py

```
@@ -0,0 +1,187 @@
+            segmentation_service = tool.segmentation_service_name
+            break
+        elif hasattr(tool, "service_name"):
+            # For tools that only use detection service
+            detection_service = tool.service_name
+            break
+
+    if detection_service is None or segmentation_service is None:
+        raise RuntimeError(
+            "Required perception tools not found in tools list. "
+            "GetObjectGrippingPointsTool or tools with service_name property required."
+        )
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)